src/ray/object_manager/object_manager.h

```diff
@@ -197,6 +197,34 @@ class ObjectManager {
   /// Cache of locally available objects.
   std::unordered_map<ObjectID, ObjectInfoT, UniqueIDHasher> local_objects_;
 
+  /// Lock for concurrent writes to in_transit_receives_.
+  std::mutex in_transit_receives_lock;
+
+  // TODO(hme): Also suppress Push for objects already in transit.
+  /// Objects that are currently being received.
+  std::unordered_set<ObjectID, UniqueIDHasher> in_transit_receives_;
+
+  /// Record an object receive as soon as one of its chunks begins
+  /// being received.
+  void AddObjectInTransit(const ObjectID &object_id){
```

Comment:
Google's C++ style [guide](https://google.github.io/styleguide/cppguide.html) says to put all methods before data members.

Reply:
I think it makes sense to do this in a separate PR, so as to not obscure the changes in this PR.